crates/std_detect/src/detect/os/x86.rs

```diff
@@ -249,5 +249,25 @@ pub(crate) fn detect_features() -> cache::Initializer {
         }
     }
 
+    // Unfortunately, some Skylake chips erroneously report support for BMI1 and
+    // BMI2 without actual support. These chips don't support AVX, and it seems
+    // that all Intel chips with non-erroneous support BMI do (I didn't check
+    // other vendors), so we can disable these flags for chips that don't also
+    // report support for AVX.
+    //
+    // It's possible this will pessimize future chips that do support BMI and
+    // not AVX, but this seems minor compared to a hard crash you get when
+    // executing an unsupported instruction (to put it another way, it's safe
+    // for us to under-report CPU features, but not to over-report them). Still,
+    // to limit any impact this may have in the future, we only do this for
+    // Intel chips, as it's a bug only present in their chips.
+    //
+    // This bug is documented as `SKL052` in the errata section of this document:
+    // http://www.intel.com/content/dam/www/public/us/en/documents/specification-updates/desktop-6th-gen-core-family-spec-update.pdf
+    if vendor_id == *b"GenuineIntel" && !value.test(Feature::avx as u32) {
+        value.unset(Feature::bmi1 as u32);
+        value.unset(Feature::bmi2 as u32);
+    }
```

Comment:
Could you limit this to the specific cpu family that has the bug?

Reply:
Probably in theory, I mean there's model/family info in the bits of __cpuid_count(1, 0).eax...

But it's not clear to me at all which values for these correspond to the impacted chip :(

Reply:
Affected Steppings for this appear to be...
| Number | Y  | U | - | H | - | S | - | 
|--|--|--|--|--|--|--|--|
| - | D-1 | D-1 | K-1 | N-0 | R-0 | R-0 | S-0 |
| SKL052 | X | X | X | X | X | X | X |

So that looks like effectively "all Skylake (non-server)", according to Intel, therefore we can constrain this logic to all Skylake processors. We don't know which of the processors this is actually true for, since it's a "may", so we must apply it to all Skylake "client" chips. The possible values for family, model, etc., are specified on page 21 of this specification update, so we just add another constraint.

New Skylake chips are not being shipped so we can assume this won't change.